course_info/tests/jasmine/directives/BadgeDirectiveSpec.js

```diff
@@ -0,0 +1,45 @@
+describe('Unit testing BadgeDirective', function() {
+    beforeEach(module('CourseInfo'));
+
+    var $compile, $rootScope;
+    beforeEach(inject(function(_$compile_, _$rootScope_) {
+        $compile = _$compile_;
+        $rootScope = _$rootScope_;
+    }));
+
+    // sanity check the test framework
+    it('should inject the providers we requested', function() {
+        [$compile, $rootScope].forEach(function(thing) {
+            expect(thing).not.toBeUndefined();
+            expect(thing).not.toBeNull();
+        });
+    });
+
+    // these are just copied from BadgeDirective.js, and should be kept in sync
+    var roleToBadge = {
+        CLASPART: 'HUID',
+        COUNTWAY: 'LIBRARY',
+        EMPLOYEE: 'HUID',
+        STUDENT: 'HUID',
+        WIDENER: 'LIBRARY',
+        XIDHOLDER: 'XID',
+    };
+    var badgeToClass = {
+        HUID: 'label-danger',
+        XID: 'label-primary',
+        LIBRARY: 'label-success',
+        OTHER: 'label-warning',
+    };
+
+    for (role in roleToBadge) {
```

Comment:
Perhaps one more test for when the role isn't in `roleToBadge` (i.e. test for `OTHER`)